src/pages/schools/components/SchoolModal.jsx

```diff
@@ -0,0 +1,164 @@
+    form: {
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "space-around",
+        height: "100%",
+    },
+    centerItem: theme.styles.centerItem,
+});
+
+class SchoolModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            school: {},
+            submit: {},
+            errors: {
+                name: "",
+                head: "",
+            },
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.schoolId !== null && this.props.schoolId) {
+            this.props.FetchSchool(this.props.schoolId);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.props.selectedSchool !== prevProps.selectedSchool) {
+            this.setState({
+                school: { ...this.props.selectedSchool } ?? {},
+            });
+
+            if (prevProps.selectedSchool && this.props.selectedSchool) {
+                const action = this.props.schoolId ? "updated" : "created";
+                toast.success(`School ${this.props.selectedSchool.name} ${action} successfully 🙌`, {
+                    position: toast.POSITION.TOP_CENTER,
+                });
+            }
+        }
+    }
+
+    validateInput = (event) => {
+        event.preventDefault();
+
+        let errors = this.state.errors;
+        let { name, value } = event.target;
+
+        // validate input
+        errors[name] = value ? "" : (name === "name" ? "School" : "School Head") + " Name cannot be empty";
+
+        this.setState({ errors });
+    };
+
+    handleInputChange = (event) => {
+        let school = this.state.school;
+        let submit = this.state.submit;
+        let { name, value } = event.target;
+
+        // validate new input value
+        this.validateInput(event);
+        // update value in submit if it is valid
+        if (!this.state.errors[name]) {
+            submit[name] = value;
+        }
+
+        school[name] = value;
+        this.setState({ school, submit });
+    };
+
+    handleSubmit = (event) => {
+        event.preventDefault();
+
+        const { submit, errors } = this.state;
+
+        // make sure there are no errors
+        if (errors.name || errors.head) return;
+
+        const { schoolId, selectedSchool } = this.props;
+        const submit_keys = Object.keys(submit);
+
+        if (schoolId) {
+            if (submit_keys.length && !submit_keys.every((key) => selectedSchool[key] === submit[key])) {
+                this.props.UpdateSchool(submit);
+            } else {
+                toast.error("Please update some information 😓", {
+                    position: toast.POSITION.TOP_CENTER,
+                });
+            }
+        } else {
+            if (submit.hasOwnProperty("name") && submit["name"]) {
+                this.props.CreateSchool(submit);
+            } else {
+                toast.error("Please add email, type and password 😓", {
```

Comment:
What is this for?

Reply:
hmm, copy pasting is never a good idea

Reply:
also why hasOwnProperty?

Reply:
that entire part is now redundant tbh, since I'm validating all inputs now anyway. This was just a final check before submitting to make sure there is some form of validation available